src/util/RouterUtil.js

```diff
@@ -124,7 +124,24 @@ function (Okta, Util, OAuth2Util, Enums, BrowserFeatures, Errors, ErrorCodes) {
             }));
           }
         }
-      });
+      };
+
+      if (res.type === Enums.SESSION_STEP_UP && res._embedded.target &&
+          res._embedded.target.type === Enums.TARGET_APP) {
```

Comment:
Also, I think in general this code will be nicer if we're not defining the successType and then redefining it. I.e. something like this:

```javascript
const isStepUp = booleanCondition && stuff_below;

var successData = { stuff that is common };

if (isStepUp) {
  successData.stepUp = {
    url: targetUrl,
    redirect: function () {}
  }
}
else {
  successData.session = { foo }
}
```